core/spec/lib/spree/event/test_interface_spec.rb

```diff
@@ -0,0 +1,193 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'spree/event/adapters/deprecation_handler'
+require 'spree/event/test_interface'
+
+RSpec.describe Spree::Event::TestInterface do
+  let(:counter) do
+    Class.new do
+      attr_reader :count
+
+      def initialize
+        @count = 0
+      end
+
+      def inc
+        @count += 1
+      end
+    end
+  end
+
+  describe '#enable_test_interface' do
+    context 'when using the legacy adapter' do
+      it 'raises an error' do
+        adapter = Spree::Config.events.adapter
+        Spree::Config.events.adapter = Spree::Event::Adapters::ActiveSupportNotifications
+
+        expect {
+          Spree::Event.enable_test_interface
+        }.to raise_error(/test interface is not supported/)
+      ensure
+        Spree::Config.events.adapter = adapter
+      end
+    end
+  end
+
+  unless Spree::Event::Adapters::DeprecationHandler.legacy_adapter?
+    describe '#performing_only' do
+      before { Spree::Event.enable_test_interface }
+
+      it 'only performs given listeners for the duration of the block' do
+        dummy1, dummy2, dummy3 = Array.new(3) { counter.new }
+        listener1 = Spree::Event.subscribe('foo') { dummy1.inc }
+        listener2 = Spree::Event.subscribe('foo') { dummy2.inc }
+        listener3 = Spree::Event.subscribe('foo') { dummy3.inc }
+
+        Spree::Event.performing_only(listener1, listener2) do
+          Spree::Event.fire('foo')
+        end
+
+        aggregate_failures do
+          expect(dummy1.count).to be(1)
+          expect(dummy2.count).to be(1)
+          expect(dummy3.count).to be(0)
+        end
+      ensure
+        Spree::Event.unsubscribe(listener1)
+        Spree::Event.unsubscribe(listener2)
+        Spree::Event.unsubscribe(listener3)
```

Comment:
I'm not 100% sure we need those `ensure` blocks: I think that the `before` block from above by calling `Spree::Event.enable_test_interface` sets the adapter to a throwaway one, is that correct?

Reply:
Good point. `#enable_test_interface` only includes some methods to the module, but `#performing_only` has its own `ensure` block to, as you said, restore the previous adapter. So there's really no point for them in the test file. I removed all of them, besides the ones dealing with the subscriber registry.